From ee58cf71b95e69ea7534932b60a3f6e0f8e82928 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 22 Dec 2015 16:57:06 +0800
Subject: [PATCH] MLK-12043: ASoC: fsl: imx-cs42888: Add return value check

Report by coverity (CID 18428). The return value need be checked
for snd_soc_dai_set_sysclk().

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 382404d748b903cef8b87855e0f12ce5de2d2d79)
---
 sound/soc/fsl/imx-cs42888.c | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-cs42888.c b/sound/soc/fsl/imx-cs42888.c
index e097fc2..1b912c0 100644
--- a/sound/soc/fsl/imx-cs42888.c
+++ b/sound/soc/fsl/imx-cs42888.c
@@ -55,12 +55,22 @@ static int imx_cs42888_surround_hw_params(struct snd_pcm_substream *substream,
 	dai_format = SND_SOC_DAIFMT_LEFT_J | SND_SOC_DAIFMT_NB_NF |
 		     SND_SOC_DAIFMT_CBS_CFS;
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
-		snd_soc_dai_set_sysclk(cpu_dai, ESAI_HCKT_EXTAL,
+		ret = snd_soc_dai_set_sysclk(cpu_dai, ESAI_HCKT_EXTAL,
 			       priv->mclk_freq, SND_SOC_CLOCK_OUT);
 	else
-		snd_soc_dai_set_sysclk(cpu_dai, ESAI_HCKR_EXTAL,
+		ret = snd_soc_dai_set_sysclk(cpu_dai, ESAI_HCKR_EXTAL,
 			       priv->mclk_freq, SND_SOC_CLOCK_OUT);
-	snd_soc_dai_set_sysclk(codec_dai, 0, priv->mclk_freq, SND_SOC_CLOCK_IN);
+	if (ret) {
+		dev_err(dev, "failed to set cpu sysclk: %d\n", ret);
+		return ret;
+	}
+
+	ret = snd_soc_dai_set_sysclk(codec_dai, 0,
+				priv->mclk_freq, SND_SOC_CLOCK_IN);
+	if (ret) {
+		dev_err(dev, "failed to set codec sysclk: %d\n", ret);
+		return ret;
+	}
 
 	/* set cpu DAI configuration */
 	ret = snd_soc_dai_set_fmt(cpu_dai, dai_format);
-- 
1.8.0

